### PR TITLE
Fix ksa example

### PR DESCRIPTION
**Pull request by mathieu-benoit**
TF file are auto-generated based on the YAML files, so cleanup this up otherwise here https://developer.humanitec.com/examples/resource-definitions/template-driver/serviceaccount/, on the TF tab, we see 4 files instead of 2 files.

### CODE REVIEW

File: resource-definitions/template-driver/serviceaccount/serviceaccount-k8ssa-def.tf [hunk 1]
```diff
@@ -7,7 +7,7 @@ resource "humanitec_resource_definition" "serviceaccount-k8s-service-account" {
     values_string = jsonencode({
       "res_id" = "$${context.res.id}"
       "templates" = {
-        "init"      = "name: {{ index (splitList \".\" \"$${context.res.id}\") 1 }}\n"
+        "init"      = "name: {{ index ( .driver.values.res_id | splitList \".\" ) 1 }}\n"
```
Comment (mathieu-benoit) [inline resource-definitions/template-driver/serviceaccount/serviceaccount-k8ssa-def.tf:10]: Actually use the `res_id`.

File: resource-definitions/template-driver/serviceaccount/serviceaccount-k8ssa-def.yaml [hunk 2]
```diff
@@ -11,9 +11,8 @@ entity:
     values:
       res_id: ${context.res.id}
       templates:
-        # Name the ServiceAccount after the Resource
         init: |
-          name: {{ index (splitList "." "${context.res.id}") 1 }}
+          name: {{ index ( .driver.values.res_id | splitList "." ) 1 }}
```
Comment (mathieu-benoit) [inline resource-definitions/template-driver/serviceaccount/serviceaccount-k8ssa-def.yaml:15]: Actually use the `res_id`.

File: resource-definitions/template-driver/serviceaccount/README.md [hunk 3]
```diff
@@ -14,9 +14,4 @@ flowchart LR
   workload --> serviceAccount(id: modules.my-workload\ntype: k8s-service-account\nclass: default)
 ```
 
-Note that the resource `id` is used in the `k8s-service-account` Resource Definition to derive the name of the actual Kubernetes ServiceAccount. Check the code for details.
-
-Example files:
```
Comment (mathieu-benoit) [inline resource-definitions/template-driver/serviceaccount/README.md:19]: It's implicit, files will be listed, we don't need to repeat this part.

**Review by TobiasBabin — APPROVED** [review]
LGTM!

**Pull request merged**